pkg/reference/enqueue_annotations.go

```diff
@@ -131,10 +131,10 @@ func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueu
 // Annotations are accumulated as a comma-separated list.
 func Add(owner types.NamespacedName, object client.Object) error {
 	if owner.Namespace == "" {
-		return fmt.Errorf("%T does not have a namespace, cannot call Add", owner)
+		return fmt.Errorf("%T does not have a namespace, cannot call Add", owner) //nolint:goerr113 //reason useful to have owner in error
 	}
 	if owner.Name == "" {
-		return fmt.Errorf("%T does not have a name, cannot call Add", owner)
+		return fmt.Errorf("%T does not have a name, cannot call Add", owner) //nolint:goerr113 //reason useful to have owner in error
```

Comment:
Does this check only allow static errors? A Error creating function isn't allowed in some way for dynamic data?

Reply:
Yeah, func creating error should work. Reason I didn't touch this code much is that it's "ported" from operator-lib.